13.0 mig partner contact in several companies

foliveira-odoogap opened the pull request:
[MIG] 13.0 partner_contact_in_several_companies

Commit:
Deprecate base_contact.

- Remove nationality and birthday features.
- Move the remaining features to partner_contact_in_several_companies.
- Make base_contact a dummy module.
- Warn deprecation everywhere possible.

Commit:
Deprecate base_contact.

- Remove nationality and birthday features.
- Move the remaining features to partner_contact_in_several_companies.
- Make base_contact a dummy module.
- Warn deprecation everywhere possible.

sysadminmatmoz committed:
Translations and templates sync

sysadminmatmoz committed:
Translations and templates sync

yvaucher committed:
Add missing default oca icons

yvaucher committed:
Add missing default oca icons

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

Commit:
OCA module upgrade to version 9.0.

- Remove Domain Error on contact view.
- Additionally, improve behaviour of personal contact info page.
- Change to README.
- No longer dependent on partner_contact_personal_information_page
- Added tests for ir_actions
- Include extra demo data.
- Increase test coverage.
- Remove birthdate from demo data.

Commit:
OCA module upgrade to version 9.0.

- Remove Domain Error on contact view.
- Additionally, improve behaviour of personal contact info page.
- Change to README.
- No longer dependent on partner_contact_personal_information_page
- Added tests for ir_actions
- Include extra demo data.
- Increase test coverage.
- Remove birthdate from demo data.

yvaucher committed:
Making the 'personal contact' informations depentens of the module 'personal_contact_information'

yvaucher committed:
Making the 'personal contact' informations depentens of the module 'personal_contact_information'

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

pedrobaeza committed:
[MIG] Rename manifest files

Commit:
[MIG] 10.0 partner_contact_in_several_companies (#347)

* [MIG] inital work to 10.0

* [FIX] restore computed field

- select=1 is not usable anymore ...

* [FIX] convert _fields_sync to new api

* [FIX] restore form . import statement

* [FIX] readme with last template.

* [FIX] import statement.

* [FIX] check all tests

* [FIX] Flake8 errors fixes

* [FIX] flake 8

* [FIX] wrong obj call self -> self.contact_id

* [FIX] typo

* [FIX] flake 8

* [FIX] add index on contact_type

* [fix] use in statement on test

* [FIX] update base action with new context

* [FIX] test fixes

* [FIX] test work

* [FIX] flake 8 over indent

* [FIX] fixes on lastest comments

- better readability
- remove <data></data> tags

* [FIX] model that same pattern for _fields_sync

- add test ensure_once()

* [FIX] remove last data xml tags

* [FIX] fix action test

* [FIX] typo in import statement

* [FIX] remove logger from tests

* [FIX] new flake8 compute method name for OCA

* [FIX] minor typo

* [FIX] partner_contact_in_several_companies: remove contributor key in manifest

* [FIX] partner_contact_in_several_companies: residual onchange on partner view

* Revert "[FIX] partner_contact_in_several_companies: remove contributor key in manifest"

This reverts commit 56e511a9a7e1dd20cf99578a340a00301a98ba85.

* [FIX] partner_contact_personal_information_page: clean manifest and authors

* [FIX] partner_contact_in_several_companies: required = True not needed

Commit:
[MIG] 10.0 partner_contact_in_several_companies (#347)

* [MIG] inital work to 10.0

* [FIX] restore computed field

- select=1 is not usable anymore ...

* [FIX] convert _fields_sync to new api

* [FIX] restore form . import statement

* [FIX] readme with last template.

* [FIX] import statement.

* [FIX] check all tests

* [FIX] Flake8 errors fixes

* [FIX] flake 8

* [FIX] wrong obj call self -> self.contact_id

* [FIX] typo

* [FIX] flake 8

* [FIX] add index on contact_type

* [fix] use in statement on test

* [FIX] update base action with new context

* [FIX] test fixes

* [FIX] test work

* [FIX] flake 8 over indent

* [FIX] fixes on lastest comments

- better readability
- remove <data></data> tags

* [FIX] model that same pattern for _fields_sync

- add test ensure_once()

* [FIX] remove last data xml tags

* [FIX] fix action test

* [FIX] typo in import statement

* [FIX] remove logger from tests

* [FIX] new flake8 compute method name for OCA

* [FIX] minor typo

* [FIX] partner_contact_in_several_companies: remove contributor key in manifest

* [FIX] partner_contact_in_several_companies: residual onchange on partner view

* Revert "[FIX] partner_contact_in_several_companies: remove contributor key in manifest"

This reverts commit 56e511a9a7e1dd20cf99578a340a00301a98ba85.

* [FIX] partner_contact_personal_information_page: clean manifest and authors

* [FIX] partner_contact_in_several_companies: required = True not needed

Commit:
[MIG] V11 migration

- fax field removed from res.partner
- 2to3 on all py files
- add contributor and update README
- use hasclass in place of @class in xpath expr
- improve tests

[FIX] typo and clean README

Commit:
[MIG] V11 migration

- fax field removed from res.partner
- 2to3 on all py files
- add contributor and update README
- use hasclass in place of @class in xpath expr
- improve tests

[FIX] typo and clean README

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] partner_contact_in_several_companies: Incorrect syntax for XML context

pedrobaeza committed:
[FIX] partner_contact_in_several_companies: Incorrect syntax for XML context

drdran committed:
[FIX] partner_contact_in_several_companies: Fix problem in view (#541)

drdran committed:
[FIX] partner_contact_in_several_companies: Fix problem in view (#541)

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[FIX] disable position editing on contact

- This is just to see all positions
- If we want to add position go to one company and add contact linked to this partner

Commit:
[FIX] disable position editing on contact

- This is just to see all positions
- If we want to add position go to one company and add contact linked to this partner

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-11.0/partner-contact-11.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-11-0/partner-contact-11-0-partner_contact_in_several_companies/es/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-11.0/partner-contact-11.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-11-0/partner-contact-11-0-partner_contact_in_several_companies/es/

kongrattapong committed:
[MIG] partner_contact_in_several_companies: Migration to 12.0

kongrattapong committed:
[MIG] partner_contact_in_several_companies: Migration to 12.0

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

oca-travis committed:
[UPD] Update partner_contact_in_several_companies.pot

OCA-git-bot committed:
partner_contact_in_several_companies 12.0.2.0.0

OCA-git-bot committed:
partner_contact_in_several_companies 12.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-12.0/partner-contact-12.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_contact_in_several_companies/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-12.0/partner-contact-12.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_contact_in_several_companies/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_contact_in_several_companies/de/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (29 of 29 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_contact_in_several_companies
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_contact_in_several_companies/de/

rousseldenis commented:
@foliveira-odoogap I think this is missing migration commit, no ? Or is it WIP ?

foliveira-odoogap commented:
> @foliveira-odoogap I think this is missing migration commit, no ? Or is it WIP ?

@rousseldenis no, it's not missing any migration commit

rousseldenis commented on partner_contact_in_several_companies/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+{
+    "name": "Contacts in several partners",
+    "summary": "Allow to have one contact in several partners",
+    "version": "12.0.2.0.0",
```
@foliveira-odoogap That's make me think you forget to embed migration commit.

foliveira-odoogap replied:
My bad, I'll correct that as soon as possible!

foliveira-odoogap replied:
@rousseldenis I embed the migration commit.

foliveira-odoogap committed:
[MIG] partner_contact_in_several_companies: Migration to 13.0

foliveira-odoogap committed:
[MIG] partner_contact_in_several_companies: Migration to 13.0

regispirard commented on partner_contact_in_several_companies/views/res_partner.xml:
```diff
@@ -1,149 +1,165 @@
-                                        <img t-if="record.image.raw_value" t-att-src="'data:image/png;base64,'+record.image.raw_value"/>
-                                        <t t-if="!record.image.raw_value">
-                                            <img t-if="record.is_company.raw_value === true" t-att-src='_s + "/base/static/src/img/company_image.png"'/>
-                                            <img t-if="record.is_company.raw_value === false" t-att-src='_s + "/base/static/src/img/avatar.png"'/>
```
company_image.png and avatar.png where moved to /base/static/img/

<img t-if="record.is_company.raw_value === true" t-att-src='_s + "/base/static/img/company_image.png"'/>
<img t-if="record.is_company.raw_value === false" t-att-src='_s + "/base/static/img/avatar.png"'/>

regispirard commented:
@foliveira-odoogap : Except for my 3 previous comments (minor changes to do), it's all good, thank you 👍

regispirard reviewed (changes requested):
@foliveira-odoogap : Except for my 3 previous comments (minor changes to do), it's all good, thank you 👍

foliveira-odoogap commented:
> @foliveira-odoogap : Except for my 3 previous comments (minor changes to do), it's all good, thank you
@regispirard Ok, I'll fix it as soon as I can!

foliveira-odoogap committed:
[IMP] fix minor changes in manyfest and view

foliveira-odoogap committed:
[IMP] fix minor changes in manyfest and view

HaraldPanten reviewed (changes requested):
Could you check travis and commets, please?

regispirard reviewed (approved):
LGTM (code + functional review)

hugho-ad committed:
[FIX] partner_second_lastname: False value should be set on lastname2
not firstname

before this commit the following error happen:
name = 'Van-Eyck Jan'

this was converted to

firstname = False
lastname= 'Van-Eyck'
lastname2 = 'Jan'

and it should be like

firstname = 'Jan'
lastname= 'Van-Eyck'
lastname2 = False

foliveira-odoogap committed:
[IMP] check travis failure

foliveira-odoogap committed:
[IMP] check travis failure

foliveira-odoogap committed:
[IMP] check travis failure_2

foliveira-odoogap committed:
[IMP] check travis failure_2

foliveira-odoogap committed:
[IMP] check travis failure_3

foliveira-odoogap committed:
[IMP] check travis failure_3

foliveira-odoogap commented:
@regispirard I'm having some troubles trying to fixing those failures, could you tell me what I need to do, please?

![Screenshot from 2020-09-24 16-19-15](https://user-images.githubusercontent.com/62010839/94165090-bfc7cc80-fe81-11ea-8a4e-004aa4068bb3.png)

rousseldenis commented:
> @regispirard I'm having some troubles trying to fixing those failures, could you tell me what I need to do, please?
> 
> ![Screenshot from 2020-09-24 16-19-15](https://user-images.githubusercontent.com/62010839/94165090-bfc7cc80-fe81-11ea-8a4e-004aa4068bb3.png)

You have to install pre-commit in the repo you are  working on before comitting. ('pre-commit install').

You have error here:

![image](https://user-images.githubusercontent.com/19529533/94239604-593ebf00-ff12-11ea-8ad1-fe77fa7a187d.png)

foliveira-odoogap committed:
[IMP] pre-commit version downgrade

foliveira-odoogap committed:
[IMP] pre-commit version downgrade

foliveira-odoogap commented:
@rousseldenis everything's ok to merge =)

regispirard commented:
> @rousseldenis everything's ok to merge =)

Good job ;-)

rousseldenis commented:
Cool, could you squash your fixes commits into the migration one ? Thanks

foliveira-odoogap commented:
> Cool, could you squash your fixes commits into the migration one ? Thanks

By squash, you mean doing rebase with my branch 13.0?

Commit:
[FIX] Fixing problems with migration of computed fields with search function and no computed function defined

OCA-git-bot committed:
Merge PR #966 into 13.0

Signed-off-by NL66278

OCA-git-bot committed:
partner_multi_relation 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_ref_unique
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_ref_unique/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_multi_relation
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_multi_relation/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_second_lastname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_second_lastname/

TDu committed:
[FIX][13.0] partner_identification id number check

OCA-git-bot committed:
Merge PR #980 into 13.0

Signed-off-by moylop260

OCA-git-bot committed:
partner_second_lastname 13.0.2.0.0

OCA-git-bot committed:
Merge PR #975 into 13.0

Signed-off-by rousseldenis

OCA-git-bot committed:
partner_identification 13.0.1.0.1

ygol committed:
Added translation using Weblate (Hebrew)

ygol committed:
Added translation using Weblate (Hebrew)

foliveira-odoogap commented:
@rousseldenis can you confirm if everything's ok to proceed?

ajaniszewska-dev committed:
[FIX][13.0] partner_identification: Add menu for Partner ID Numbers

ygol committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_fax
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_fax/he/

ygol committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (16 of 16 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_firstname/he/

rven committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_contact_birthdate
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_contact_birthdate/nl/

NL66278 commented:
@foliveira-odoogap Squashing your commits basically means that all the commits you made for the migration are merged into one commit. You have to do an interactive rebase from the commit just before your migration commit: $ git rebase -i <commit to rebase from>. You might use $ git rebase -i HEAD~5 to get the last 5 commits. You then get an editor screen with each commit on one line. You can mark all the commits after the first as a fixup, by replacing the commit hash with an 'f'. May sound complicated, but is actually quite easy.

dreispt commented:
@foliveira-odoogap Can you please squash commit, as per https://github.com/OCA/partner-contact/pull/946#issuecomment-712168420
Let me know if you need help

foliveira-odoogap commented:
> @foliveira-odoogap Can you please squash commit, as per [#946 (comment)](https://github.com/OCA/partner-contact/pull/946#issuecomment-712168420)
> Let me know if you need help

Hi @dreispt, maybe it would be better for you to help me.

OCA-git-bot committed:
Merge PR #984 into 13.0

Signed-off-by dreispt

oca-travis committed:
[UPD] Update partner_identification.pot

OCA-git-bot committed:
partner_identification 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_identification/

OCA-git-bot committed:
[IMP] pre-commit: pin node version [ci skip]

ajaniszewska-dev committed:
[13.0][IMP] Add missing field to ID Numbers tree view

OCA-git-bot committed:
Merge PR #1005 into 13.0

Signed-off-by jgrandguillaume

OCA-git-bot committed:
partner_identification 13.0.1.2.0

foliveira-odoogap commented:
> > @foliveira-odoogap Can you please squash commit, as per [#946 (comment)](https://github.com/OCA/partner-contact/pull/946#issuecomment-712168420)
> > Let me know if you need help
> 
> Hi @dreispt, maybe it would be better for you to help me.

@dreispt can you help me please?

rousseldenis commented:
@foliveira-odoogap What do you need ?

foliveira-odoogap commented:
> @foliveira-odoogap What do you need ?

I don't know what you mean by squashing my commits, can tell me how you do that please?

rousseldenis commented:
> > @foliveira-odoogap What do you need ?
> 
> I don't know what you mean by squashing my commits, can tell me how you do that please?

When developing, it's common to commit step by step, adding or removing features.

But some commits have meaning (enclosed modifications have sense) and some could have not.

So, git have feature to 'merge' some commits in the stack together.

To do so, see: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

It's an interactive rebase. You choose how many commits to rebase and then modify the operation on each of them (pick/fixup/drop/rename...).

In your case, git rebase -i HEAD~6, fixup for the five last ones and pick for the Migration one.

foliveira-odoogap committed:
Merge branch '13.0-mig-partner_contact_in_several_companies' of https://github.com/foliveira-odoogap/partner-contact into 13.0-mig-partner_contact_in_several_companies

foliveira-odoogap commented:
> > > @foliveira-odoogap What do you need ?
> > 
> > 
> > I don't know what you mean by squashing my commits, can tell me how you do that please?
> 
> When developing, it's common to commit step by step, adding or removing features.
> 
> But some commits have meaning (enclosed modifications have sense) and some could have not.
> 
> So, git have feature to 'merge' some commits in the stack together.
> 
> To do so, see: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git
> 
> It's an interactive rebase. You choose how many commits to rebase and then modify the operation on each of them (pick/fixup/drop/rename...).
> 
> In your case, git rebase -i HEAD~6, fixup for the five last ones and pick for the Migration one.

Ok, I squashes my commits and pushed them, now you have to merge right?

rousseldenis commented:
Indeed, as soon as you have pushed and tests green, I'll do it

remihb commented:
> In your case, git rebase -i HEAD~6, fixup for the five last ones and pick for the Migration one.

Yes @foliveira-odoogap please do as @rousseldenis explains.
I cloned your code and still have your 6 migration commits. Maybe you forget to push ?

HaraldPanten commented:
@foliveira-odoogap any news about this?

remihb commented:
something went wrong !

NL66278 commented:
/ocabot merge minor
